flutter/lib/src/integrations/native_app_start_handler.dart

```diff
@@ -0,0 +1,303 @@
+// ignore_for_file: invalid_use_of_internal_member
+
+import '../../sentry_flutter.dart';
+import '../native/native_app_start.dart';
+import '../native/sentry_native_binding.dart';
+
+// ignore: implementation_imports
+import 'package:sentry/src/sentry_tracer.dart';
+
+/// Handles communication with native frameworks in order to enrich
+/// root [SentryTransaction] with app start data for mobile vitals.
+class NativeAppStartHandler {
+  NativeAppStartHandler(this._native);
+
+  final SentryNativeBinding _native;
+
+  late final Hub _hub;
+  late final SentryFlutterOptions _options;
+
+  /// We filter out App starts more than 60s
+  static const _maxAppStartMillis = 60000;
+
+  Future<void> call(Hub hub, SentryFlutterOptions options,
+      {required DateTime? appStartEnd}) async {
+    _hub = hub;
+    _options = options;
+
+    final nativeAppStart = await _native.fetchNativeAppStart();
+    if (nativeAppStart == null) {
+      return;
+    }
+    final appStartInfo = _infoNativeAppStart(nativeAppStart, appStartEnd);
+    if (appStartInfo == null) {
+      return;
+    }
+
+    // Create Transaction & Span
+
+    const screenName = SentryNavigatorObserver.rootScreenName;
```

Comment:
Can we move the `rootScreenName` somewhere else? I was confused at first because I thought we had a dependency to navigator observer here

Reply:
phew, tough call, could even be in the handler if we don't need it anywhere else` _rootScreenName`

Reply:
https://github.com/getsentry/sentry-dart/blob/feat/improve-app-start-integration/flutter/lib/src/navigation/sentry_navigator_observer.dart#L250C1-L253C1

we use it here but I think we can also remove this since we won't ever do a transaction on the root screen through the navigator observer since app starts/ttid is now managed through the integration

Reply:
Do we really want to remove the root screen transactions from the observer? Maybe there is still value in having this, as the other is the whole app start, and here "only" of the root screen, no? We'd also be changing behaviour.